lib/galaxy/webapps/tool_shed/buildapp.py

```diff
@@ -277,23 +260,33 @@ def wrap_in_middleware( app, global_conf, application_stack, **local_conf ):
     app = wrap_if_allowed( app, stack, XForwardedHostMiddleware )
     # Various debug middleware that can only be turned on if the debug
     # flag is set, either because they are insecure or greatly hurt
-    # performance. The print debug middleware needs to be loaded last,
-    # since there is a quirk in its behavior that breaks some (but not
-    # all) subsequent middlewares.
+    # performance.
     if debug:
         # Middleware to check for WSGI compliance
         if asbool( conf.get( 'use_lint', True ) ):
             from paste import lint
             app = wrap_if_allowed( app, stack, lint.make_middleware, name='paste.lint', args=(conf,) )
         # Middleware to run the python profiler on each request
         if asbool( conf.get( 'use_profile', False ) ):
-            import profile
+            from paste.debug import profile
             app = wrap_if_allowed( app, stack, profile.ProfileMiddleware, args=(conf,) )
-        # Middleware that intercepts print statements and shows them on the
-        # returned page
-        if asbool( conf.get( 'use_printdebug', True ) ):
-            from paste.debug import prints
-            app = wrap_if_allowed( app, stack, prints.PrintDebugMiddleware, args=(conf,) )
+        if interactive:
+            # Interactive exception debugging, scary dangerous if publicly
+            # accessible, if not enabled we'll use the regular error printing
+            # middleware.
+            try:
+                from weberror import evalexception
+                app = wrap_if_allowed_or_fail( app, stack, evalexception.EvalException,
+                                               args=(conf,),
+                                               kwargs=dict(templating_formatters=build_template_error_formatters()) )
+            except MiddlewareWrapUnsupported as exc:
+                log.warning(str(exc))
+                import galaxy.web.framework.middleware.error
+                app = wrap_if_allowed( app, stack, galaxy.web.framework.middleware.error.ErrorMiddleware, args=(conf,) )
+    elif debug and not interactive:
+        # Not in interactive debug mode, just use the regular error middleware
+        import galaxy.web.framework.middleware.error
+        app = wrap_if_allowed( app, stack, galaxy.web.framework.middleware.error.ErrorMiddleware, args=(conf,) )
```

Comment:
this won't be reached, `debug = True` would only enter the `if debug` above. Maybe else: and indent 4.